examples/plot_vgm_error.py

```diff
@@ -0,0 +1,256 @@
+our results using `xdem.spatialstats.patches_method`, an intensive Monte-Carlo sampling approach.
+
+"""
+# sphinx_gallery_thumbnail_number = 6
+import matplotlib.pyplot as plt
+import numpy as np
+import xdem
+import geoutils as gu
+
+# %%
+# We start by loading example files including a difference of DEMs at Longyearbyen glacier and the outlines to rasterize
+# a glacier mask.
+# Prior to differencing, the DEMs were aligned using :class:`xdem.coreg.NuthKaab` as shown in
+# the :ref:`sphx_glr_auto_examples_plot_nuth_kaab.py` example. We later refer to those elevation differences as *dh*.
+
+dh = xdem.DEM(xdem.examples.get_path("longyearbyen_ddem"))
+glacier_outlines = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
+mask_glacier = glacier_outlines.create_mask(dh)
+
+# %%
+# We remove values on glacier terrain in order to isolate stable terrain, our proxy for elevation measurement errors.
+dh.data[mask_glacier] = np.nan
+
+# %%
+# We estimate the average per-pixel elevation measurement error on stable terrain, using both the standard deviation
+# and normalized median absolute deviation. For this example, we do not account for the non-stationarity in elevation
+# measurement errors quantified in :ref:`sphx_glr_auto_examples_plot_nonstationary_error.py`.
+print('STD: {:.2f}'.format(np.nanstd(dh.data))+' meters.')
+print('NMAD: {:.2f}'.format(xdem.spatialstats.nmad(dh.data))+' meters.')
+
+# %%
+# The two measures of dispersion are quite similar showing that, on average, there is a small influence of outliers on the
+# elevation differences. The per-pixel precision is about :math:`\pm` 2.5 meters.
+# **Does this mean that every pixel has an independent measurement error of** :math:`\pm` **2.5 meters?**
+# Let's plot the elevation differences to visually check the quality of the data.
+plt.figure(figsize=(8, 5))
+plt_extent = [
+    dh.bounds.left,
+    dh.bounds.right,
+    dh.bounds.bottom,
+    dh.bounds.top,
+]
+plt.imshow(dh.data.squeeze(), cmap="RdYlBu", vmin=-4, vmax=4, extent=plt_extent)
+cbar = plt.colorbar()
+cbar.set_label('Elevation differences (m)')
+plt.show()
+
+
+# %%
+# We clearly see that the residual elevation differences on stable terrain are not random. The positive and negative
+# differences (blue and red, respectively) appear correlated over large distances.
+#
+# Conclusion: **These correlated errors need to be quantified to reliably estimate elevation measurement errors for a
+# sum, or average of elevation differences samples for a specific surface area**.
+
+# %%
+# Additionally, we notice that the elevation differences are still polluted by unrealistically large elevation
+# differences near glaciers, probably because the glacier inventory is more recent than the data, and the outlines are too small.
+# To remedy this, we filter large elevation differences outside 4 NMAD.
+dh.data[np.abs(dh.data) > 4 * xdem.spatialstats.nmad(dh.data)] = np.nan
```

Comment:
Same comment as before regarding the use of NaNs. You could also possibly use np.ma functions directly like [masked_outside](https://numpy.org/doc/stable/reference/generated/numpy.ma.masked_outside.html#numpy.ma.masked_outside)

Reply:
Or you can also use the newly added "buffer" option in create_mask! :smiley: